ui/register_react_component.md

```diff
@@ -0,0 +1,98 @@
+### Register react component
+In ManageIq project there is a way of sharing components, getting their state and control their flow by something which is called ManageIq component.
+
+This registry and usage consists of three steps, define blueprint which holds the react component with it's props. Then you will define it inside shared components (use blueprint created in step 1). And when someone decides to use such component he will call function newInstance with custom props and element to which such component is bound to.
+
+This way using such component is abstract for user, he doesn't know if such component was written in pure JS, React, Angular or whatever.
+
+#### 1. Define react component
+This is your first step of creating shared abstract component. You may use our custom function which fits react, should you choose angular, Vue.js or whatever you'll have to write your own funcion.
+
+When using react blueprint function, just wrap your component (or HTML) with `addReact` function and as argument use callback which will be called with props. These props will be passed when new isntance of this component is created.
+
+```JSX
+import React from 'react';
+import addReact from 'helpers';
+import { YourComponent } from 'your-components/custom-component';
+
+addReact('YourComponentName', props => <YourComponent {...props} />);
+```
+
+#### 2. New instance
+When you want to create new instance of recently created component just call newInstance function with these arguments
+* name - name of component you wish to use.
+* props - object with props which will be mapped to your components (used when creating blueprint)
+* element - DOM element where this new instance of component should be rendered.
+```JS
+import newInstance from 'registry';
+
+const props = {};
+const element = document.getElementById('element-id');
```

Comment:
no getElementById, `newInstance` should take a normal selector

(or should this be using `componentMarkup` instead?)

Reply:
New instance expects mount to element, because we don't know how user gets such element either trough getElement, trough react prop or whatever. @Hyperkid123 is writing helper method which will be shortcut to these commands.

Reply:
What I'm saying is - the default method to instantiate a component must follow the same rules as all the other methods (like `miq_bootstrap`).

So.. if this documents `newInstance` as something you would want to use, we need to change it to take selectors.

If this documents `componentMarkup` as the thing to use, and mentions `newInstance` only for the cases where you're doing stuff manually, we don't need to change it.

But.. right now, it appears like `newInstance` is the preferred way, possibly not what you intended?